account-kit/signer/src/signer.ts

```diff
@@ -27,19 +27,30 @@ export type AuthParams =
     }
   | ({
       type: "oauth";
-      authProviderId: string;
-      isCustomProvider?: boolean;
       scope?: string;
       claims?: string;
-    } & RedirectConfig)
+    } & OauthProviderConfig &
+      OauthRedirectConfig)
   | { type: "oauthReturn"; bundle: string; orgId: string };
 
-export type OauthMode = "redirect" | "popup";
+export type OauthProviderConfig =
+  | {
+      authProviderId: "auth0";
+      isCustomProvider?: false;
+      auth0Connection?: string;
+    }
+  | {
+      authProviderId: string;
```

Comment:
should this be a union type of the providers we support? like the `keyof DEFAULT_SCOPE_AND_CLAIMS` above?

Reply:
I believe providers here can be arbitrary ids that someone has set up in their Auth0 dashboard. E.g. if someone's Auth0 has methods for SMS, Okta, Petco account, each of those would be a different id and we can't necessarily know them ahead of time.

I could maybe do some TypeScripty thing so it will have autocomplete for common ids like "github" but we do need to accept arbitrary strings here.

Reply:
ah gotcha, so this is specifically for `auth0` or does it apply to other auth providers like 1st party google and fb?

Reply:
Oh wait sorry, I thought you were talking about the `auth0connection` param, my bad.

This provider id at the moment will always be one of "google", "facebook", "apple", "auth0", but once we add custom auth providers it can be anything set from the user. Actually, I could make it so if `isCustomProvider` is false it has to be one of the known ones, that's probably a good idea.